vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/ReadPreferenceParser.java

```diff
@@ -17,40 +18,46 @@
 class ReadPreferenceParser {
   private final ReadPreference readPreference;
 
-  public ReadPreferenceParser(JsonObject config) {
-    ReadPreference rp;
-    String rps = config.getString("readPreference");
-    if (rps != null) {
-      JsonArray readPreferenceTags = config.getJsonArray("readPreferenceTags");
-      if (readPreferenceTags == null) {
-        rp = ReadPreference.valueOf(rps);
-        if (rp == null) throw new IllegalArgumentException("Invalid ReadPreference " + rps);
-      } else {
-        // Support advanced ReadPreference Tags
-        List<TagSet> tagSet = new ArrayList<>();
-        readPreferenceTags.forEach(o -> {
-          String tagString = (String) o;
-          List<Tag> tags = Stream.of(tagString.trim().split(","))
-            .map(s -> s.split(":"))
-            .filter(array -> {
-              if (array.length != 2) {
-                throw new IllegalArgumentException("Invalid readPreferenceTags value '" + tagString + "'");
-              }
-              return true;
-            }).map(array -> new Tag(array[0], array[1])).collect(Collectors.toList());
+  ReadPreferenceParser(ConnectionString connectionString, JsonObject config) {
+    ReadPreference connStringReadPreference = connectionString != null ? connectionString.getReadPreference() : null;
+    if (connStringReadPreference != null) {
+      // Prefer connection string's read preference
```

Comment:
Well, it is not big change, but it appears big since it add braces and indents for exiting code.